src/compression_wrapper.c

```diff
@@ -929,12 +929,16 @@ cr_close(CR_FILE *cr_file, GError **err)
     }
 
     if (cr_file->stat) {
+        if (cr_file->stat->hdr_checksum) {
+            free(cr_file->stat->hdr_checksum);
```

Comment:
I think this is not the right place to free the `hdr_checksum`, it should be freed together with the rest of the stat struct, ideally by `cr_contentstat_free`. Its generally not bound to the `cr_file`, its managed on its own. In fact when running with zchunk (createrepo_c --zck .) this call is causing a double free. 

I have looked around where the `hdr_checksum` could be leaking and I believe that the problem is here: https://github.com/rpm-software-management/createrepo_c/blob/master/src/repomd.c#L199
It should be using the `cr_contentstat_free` function.

Otherwise looks good to me. :slightly_smiling_face:

Reply:
fixed!